### DISCUSSION TITLE
Quarkus constructor injection. Is public constructor worse than package-private?

### DISCUSSION THREAD

**Discussion by konradgab** [quarkusio/quarkus#26651] [Q&A] [answered]
Hi
I have a noob question :(
In quarkus doc (section about CDI) is written:
"A package-private constructor injection. @Inject is optional in this particular case." 
so it is recommended to use package-private constructor injection.
Is public constructor worse or only private is worse(reflection)?
If public is worse than package-private plz explain me why
My question is only about performance not accessibility.

[a comment by an automated account is not shown]

**Comment by manovotn** [comment, marked as answer]
Hello

Public and package-private are both equally good performance wise.
Their main difference is in accessibility.
The only perf issue is with `private` due to reflection.

**Reply by konradgab** [reply]
Thanks, resolved B)
